src/Access/LDAPClient.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#if __has_include("config_core.h")
+#include "config_core.h"
+#endif
```

Comment:
#if !defined(ARCADIA_BUILD)
#include "config_core.h"
#endif

Reply:
Done.

Reply:
Yandex synchronization check:
```
could not resolve include file: ldap.h included from here: $S/clickhouse/src/Access/LDAPClient.h
```

Reply:
The only place `ldap.h` is included is wrapped with `#if USE_LDAP`. `USE_LDAP` is undefined if `ENABLE_LDAP` is evaluated to false during cmake configuration step (I verified this). How can I exactly reproduce this "Yandex synchronization check" environment?

Reply:
Yandex synchronization check is built with `ARCADIA_BUILD` defined.

Reply:
> How can I exactly reproduce this "Yandex synchronization check" environment?
@abyss7

Reply:
> > How can I exactly reproduce this "Yandex synchronization check" environment?
> 
> Probably you can't reproduce it. Maybe @abyss7 knows better.

You can't reproduce this exact problem - it's internal to us. I (or @vitlibar as well) should fix this issue in our internal repo before merging this PR.